src/foam/nanos/fs/FileUpdateDecorator.js

```diff
@@ -19,9 +20,11 @@ foam.CLASS({
       javaCode: `
         Object id = obj.getProperty("id");
         FObject oldObj = getDelegate().find(id);
-        boolean isCreate = id == null || oldObj == null;
-
-        if ( ! isCreate ) return obj;
+        
+        // Restrict file update unless lifecycleState changed
+        if ( oldObj != null && File.LIFECYCLE_STATE.compare(obj, oldObj) == 0 ) {
+          return obj;
+        }
```

Comment:
We need to protect update on the data, filename, ... 
Can we instead create a new File object with just the id and lifecycle change, and 'put' this instead.

Reply:
Does this changes allow to update filename if lifecycle changed as well?

Reply:
Updated in https://github.com/kgrgreer/foam3/pull/1566/commits/3ab39156cb38cad904f41b7cf2169418ed251a72, creating new file object with just the id and lifecycleState will nullify other properties, will need to clone instead.

> Does this changes allow to update filename if lifecycle changed as well?

No, we still restrict file update on filename, data and other properties but will allow for lifecycleState change to support file delete or possibly reactivation if requested.